File: LeetCode/2053. Kth Distinct String in an Array/solution.cpp
/** 
 * Runtime: 20 ms
 * Memory Usage: 14.2 MB 
 */
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        unordered_map<string, int> cnt;
        for (auto &num: arr) {
            cnt[num]++;
        }
        for (auto &num: arr) {
            if (cnt[num] == 1) {
                --k;
                if (k == 0) {
                    return num;
                }
            }
        }
        return "";
    }
};

File: LeetCode/2053. Kth Distinct String in an Array/solution.py
""" 
Runtime: 44 ms
Memory Usage: 15.2 MB
"""
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        cnt = Counter(arr)
        for num in arr:
            if cnt[num] == 1:
                k -= 1
                if k == 0:
                    return num
        return ''
